AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  ProximityHandler.swift
+//  AlpsSDK
+//
+//  Created by Wen on 26.10.17.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import Alps
+
+protocol ProximityHandlerDelegate : class {
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple])
+}
+
+class ProximityHandler: NSObject, ProximityHandlerDelegate {
+    var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    var beaconsDetected: [String: [IBeaconTriple]] = [:]
+    var beaconsTriggered: [String: [String: ProximityEvent]] = [:]
+    
+    override init() {
+        super.init()
+        beaconsTriggered["unknown"] = [String: ProximityEvent]()
+        beaconsTriggered["immediate"] = [String: ProximityEvent]()
+        beaconsTriggered["far"] = [String: ProximityEvent]()
+        beaconsTriggered["near"] = [String: ProximityEvent]()
+    }
+    
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple]) {
+        let unknownBeacons = beacons.filter {$0.proximity == CLProximity.unknown}
+        let immediateBeacons = beacons.filter {$0.proximity == CLProximity.immediate}
+        let nearBeacons = beacons.filter {$0.proximity == CLProximity.near}
+        let farBeacons = beacons.filter {$0.proximity == CLProximity.far}
+        
+        // Filter with existing beacons in application
+        beaconsDetected["unknown"] = synchronizeBeacons(beacons: unknownBeacons, knownBeacons: knownBeacons)
+        beaconsDetected["immediate"] = synchronizeBeacons(beacons: immediateBeacons, knownBeacons: knownBeacons)
+        beaconsDetected["near"] = synchronizeBeacons(beacons: nearBeacons, knownBeacons: knownBeacons)
+        beaconsDetected["far"] = synchronizeBeacons(beacons: farBeacons, knownBeacons: knownBeacons)
+        
+        // triggers for proximity event
+        beaconsDetected.forEach { (key, value) in
+            handleProximity(key: key, value: value)
+        }
+    }
+    
+    private func handleProximity(key: String, value: [IBeaconTriple]) {
```

Comment:
something's wrong with the method name. I can't figure out what it does looking on name only.